lib/extras/packed_image.h

```diff
@@ -268,6 +272,7 @@ class ChunkedPackedFrame {
   void* opaque_;
   ReadLine read_line_;
   std::set<void*> buffers_;
+  std::mutex* mtx_;
```

Comment:
I don't see why this should be an externally-owned mutex...

Reply:
First I added mutex here but then in complained for the place where we emplace_back a chunked frame.

Reply:
Ah, because it makes the struct not movable...

I'd perhaps make this a unique_ptr<mutex> instead, that feels safer

Reply:
Done.